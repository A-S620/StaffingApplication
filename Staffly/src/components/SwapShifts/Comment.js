import React from 'react';
import {makeStyles,} from '@material-ui/core/styles';
import {TextField} from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
    root: {
        '& .MuiTextField-root': {
            margin: theme.spacing(1),
            width: '25ch',
        },
    },
}));
export default function Comment() {
    const classes = useStyles();
    const [value, setValue] = React.useState('');

    const handleChange = (event) => {
        setValue(event.target.value);
        console.log(value)
    };
    return (
        <div>
            <TextField

                style={{width: 600}}
                id="Comment"
                label="Comment"
                multiline
                rows={4}
                variant="outlined"
                value={value}
                onChange={(event) => {
                    setValue(event.target.value);
                    console.log("value   " + value)

                }}
            />
        </div>
    );

}
