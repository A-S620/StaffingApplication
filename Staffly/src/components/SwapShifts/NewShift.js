import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {List, ListItemText, ListItemSecondaryAction, ListItem, ListSubheader} from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
    root: {
        width: 200,
        maxWidth: 360,
        backgroundColor: theme.palette.background.paper,
        position: 'relative',
        overflow: 'auto',
        maxHeight: 250,
    },
    listSection: {
        backgroundColor: 'inherit',
    },
    ul: {
        backgroundColor: 'inherit',
        padding: 0,
    },
}));

export default function NewShift() {
    const classes = useStyles();
    const [checked, setChecked] = React.useState([1]);

    const handleToggle = (value) => () => {
        const currentIndex = checked.indexOf(value);
        const newChecked = [...checked];

        if (currentIndex === -1) {
            newChecked.push(value);
        } else {
            newChecked.splice(currentIndex, 1);
        }

        setChecked(newChecked);
    };

    return (
        <List className={classes.root} subheader={<li />}>
            {[0, 1, 2, 3, 4].map((sectionId) => (
                <li key={`section-${sectionId}`} className={classes.listSection}>
                    <ul className={classes.ul}>
                        <ListSubheader>{`I'm sticky ${sectionId}`}</ListSubheader>
                        {[0, 1, 2].map((item) => (
                            <ListItem key={`item-${sectionId}-${item}`}>
                                <ListItemText primary={`Item ${item}`} />
                            </ListItem>
                        ))}
                    </ul>
                </li>
            ))}
        </List>
    );
}