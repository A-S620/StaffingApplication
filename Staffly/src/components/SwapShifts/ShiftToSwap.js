import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import {Divider, Grid, List, ListItem, ListItemText} from '@material-ui/core'

const useStyles = makeStyles((theme) => ({
    root: {
        width: 300,
        maxWidth: 360,
        backgroundColor: theme.palette.background.paper,
    },
}));

function ListItemLink(props) {
    return <ListItem button component="a" {...props} />;
}

export default function ShiftToSwap() {
    const classes = useStyles();

    return (
        <Grid item>
            <List className={classes.root} component="nav" aria-label="main mailbox folders">
                <ListItem button>
                    Test
                </ListItem>
                <ListItem button>
                    Test2
                </ListItem>
            </List>
        </Grid>
    );
}

