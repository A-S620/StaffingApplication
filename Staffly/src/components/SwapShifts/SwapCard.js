import React from 'react';
import {makeStyles} from '@material-ui/core/styles';
import UserShifts from "./UserShifts";
import NewShift from "./NewShift";
import Comment from "./Comment";
import {Box, Button, CardActions, CardContent, Paper} from "@material-ui/core";

const useStyles = makeStyles({
    root: {
        minWidth: 700,
        minHeight: 700,
    },
    bullet: {
        display: 'inline-block',
        margin: '0 2px',
        transform: 'scale(0.8)',
    },
    title: {
        fontSize: 14,
    },
    pos: {
        marginBottom: 12,
    },
});

export default function SwapCard() {
    const classes = useStyles();

    return (
        <Paper display="flex" justifyContent="center" orientation={"vertical"}>
            <CardContent>
                <Box display="flex" justifyContent="center" orientation={"vertical"}>
                    <UserShifts/>
                    <NewShift/>
                </Box>
                <Box style={{marginTop: 30}}>
                    <Comment/>
                </Box>


            </CardContent>
            <CardActions style={{justifyContent: 'center'}}>

                <Button variant="outlined" color="primary">
                    Submit
                </Button>

            </CardActions>
        </Paper>
    );
}
