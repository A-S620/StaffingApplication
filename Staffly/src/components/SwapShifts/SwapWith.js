import React from 'react';
import {Grid, makeStyles, Typography} from '@material-ui/core';
import ShiftToSwap from "./ShiftToSwap";
import NewShift from "./NewShift";

const useStyles = makeStyles((theme) => ({
    root: {
        display: 'flex',
        '& > *': {
            margin: theme.spacing(1),
            width: theme.spacing(50),
            height: theme.spacing(50),
        },
    },
    paper: {
        padding: theme.spacing(2),
        textAlign: 'center',
        color: theme.palette.text.secondary,
    },
    container: {
        maxHeight: '100%',
        width: '100%',
    },
    TextField: {
        marginTop: "30px",
        padding: "40px"
    }
}));

export default function SwapWith() {
    const classes = useStyles();

    return (
        <Grid
            container
            direction="column"
            justify="center"
            alignItems="center"
        >
            <Grid item>
                <Typography variant="h6" gutterBottom justify="center">
                    Select Shift to swap with:
                </Typography>

            </Grid>
            <Grid item>
                <NewShift/>
            </Grid>


        </Grid>


    );
}