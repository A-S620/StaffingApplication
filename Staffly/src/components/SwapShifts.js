import React from 'react';
import {Grid} from "@material-ui/core";
import SwapCard from "./SwapShifts/SwapCard";

export default function SwapShifts() {
    return (
        <Grid style={{marginTop: 10}}
              container
              direction="row"
              justify="center"
              alignItems="center"
        >
            <Grid
                container
                direction="column"
                justify="flex-start"
                alignItems="center"
                style={{marginBottom: 20}}
            >
                <SwapCard/>
            </Grid>


        </Grid>
    )
}

