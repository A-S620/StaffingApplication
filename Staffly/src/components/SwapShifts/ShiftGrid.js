import React from 'react';
import {Grid, makeStyles} from '@material-ui/core';
import SwapWith from "./SwapWith";
import UserShifts from "./UserShifts";
const useStyles = makeStyles((theme) => ({
    root: {
        display: 'flex',
        '& > *': {
            margin: theme.spacing(1),
            width: theme.spacing(50),
            height: theme.spacing(50),
        },
    },
    paper: {
        padding: theme.spacing(2),
        textAlign: 'center',
        color: theme.palette.text.secondary,
    },
    container: {
        maxHeight: '100%',
        width: '100%',
    },
    TextField: {
        marginTop: "30px",
        padding: "40px"
    }
}));

export default function ShiftGrid() {
    const classes = useStyles();

    return (
        <Grid
            container
            direction="row"
            justify="center"
            alignItems="center"
        >
            <Grid item>
                <UserShifts/>
            </Grid>
            <Grid item>
                <SwapWith/>
            </Grid>


        </Grid>


    );
}