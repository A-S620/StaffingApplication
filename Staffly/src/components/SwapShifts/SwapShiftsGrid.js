import React from 'react';
import {Box, Grid, makeStyles} from '@material-ui/core';
import ShiftGrid from "./ShiftGrid";

const useStyles = makeStyles((theme) => ({
    root: {
        display: 'flex',
        '& > *': {
            margin: theme.spacing(1),
            width: theme.spacing(50),
            height: theme.spacing(50),
        },
    },
    paper: {
        padding: theme.spacing(2),
        textAlign: 'center',
        color: theme.palette.text.secondary,
    },
    container: {
        maxHeight: '100%',
        width: '100%',
    },
    TextField: {
        marginTop: "30px",
        padding: "40px"
    }
}));

export default function SwapShiftsGrid() {
    const classes = useStyles();

    return (
        <Box
            display="flex"
            flexWrap="nowrap"
            alignItems="center"
            justify="center"
        >
            <ShiftGrid/>
        </Box>






    );
}