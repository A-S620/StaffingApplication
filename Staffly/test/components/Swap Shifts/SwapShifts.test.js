import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux'
import SwapShifts from "../../../src/components/SwapShifts";
describe('App ', () => {
    const component = shallow(<SwapShifts/>);
    it('should render correctly', () => {


        expect(component).toMatchSnapshot();
    });
    it('should have a user shift selector', () =>{
        const selector = component.find('select').at(0)
        expect(selector).toHaveLength(1)
    })
});
