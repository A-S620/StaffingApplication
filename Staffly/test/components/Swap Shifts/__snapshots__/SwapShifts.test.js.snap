// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App  should render correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="row"
  justify="center"
  style={
    Object {
      "marginTop": 10,
    }
  }
>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    direction="column"
    justify="flex-start"
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <SwapCard />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
